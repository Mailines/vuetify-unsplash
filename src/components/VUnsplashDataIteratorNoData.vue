<template>
  <v-card
    class="elevation-0 overflow-y-auto"
    :height="imageHolderHeight - 10 + 'px'"
  >
    <v-layout
      class="d-flex justify-center align-center"
      :fill-height="imageHolderHeight + 'px'"
    >
      <v-alert
        class="text-center"
        text
        outlined
        :color="accentColor"
        :style="noMatchingPictures ? { opacity: '1' } : { opacity: '0' }"
        >{{ noMatchingPicturesLabel }}</v-alert
      >
    </v-layout>
  </v-card>
</template>
<script>
export default {
  props: {
    imageHolderHeight: {
      type: Number,
      required: true
    },
    accentColor: {
      type: String,
      required: true
    },
    noMatchingPictures: {
      type: Boolean,
      required: true
    },
    noMatchingPicturesLabel: {
      type: String,
      required: true
    }
  }
};
</script>
