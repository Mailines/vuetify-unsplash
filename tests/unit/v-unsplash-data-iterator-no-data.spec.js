import VUnsplashDataIteratorNoData from "../../src/components/VUnsplashDataIteratorNoData";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";

describe("VUnsplashDataIteratorNoData.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashDataIteratorNoData, {
      localVue,
      vuetify,
      propsData: {
        imageHolderHeight: 700,
        noMatchingPictures: true,
        noMatchingPicturesLabel: "No matching pictures were found.",
        accentColor: "#1976d2"
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
