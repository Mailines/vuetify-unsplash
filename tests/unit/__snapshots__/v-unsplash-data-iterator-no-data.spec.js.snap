// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashDataIteratorNoData.vue renders and match snapshot 1`] = `
<div class="elevation-0 overflow-y-auto v-card v-sheet theme--light" style="height: 690px;">
  <div class="layout d-flex justify-center align-center fill-height">
    <div role="alert" class="v-alert text-center v-sheet v-sheet--outlined theme--light v-alert--outlined v-alert--text" style="color: rgb(25, 118, 210); caret-color: #1976d2; opacity: 1;">
      <div class="v-alert__wrapper">
        <div class="v-alert__content">No matching pictures were found.</div>
      </div>
    </div>
  </div>
</div>
`;
